internal/engine/eval/vulncheck/vulncheck.go

```diff
@@ -134,6 +144,34 @@ func (e *Evaluator) Eval(ctx context.Context, pol map[string]any, res *engif.Res
 	return nil
 }
 
+// getPatchedVersion returns a version that patches all known vulnerabilities. If no such version exists, it returns
+// the version that patches the most vulnerabilities. If none of the vulnerabilities have patches, it returns the
+// empty string.
```

Comment:
I thought this was interesting (not saying necessarily wrong) that there could be a lower package that fixes more CVEs than a newer version. Do you think that's an artifact of how fixes are tracked? Are there many cases where we'd suggest a version that's not the latest?

Reply:
We suggest a version that's not the latest if an earlier version still fixes the vulnerability, to keep with the "smallest delta" update. The example that I was testing with is `github.com/cilium/cilium` where if we're already on version `1.13.0` we can update to `1.13.7`, instead of the latest, which is `1.14.2`

Reply:
OK, in the case there are still vulnerabilities in the later versions, we'll just detect them later right? That's fine!